src/main/interface/cli.c

```diff
@@ -2229,6 +2239,93 @@ static void cliName(char *cmdline)
     printName(DUMP_MASTER, pilotConfig());
 }
 
+#if defined(USE_BOARD_INFO)
+
+#define ERROR_MESSAGE "Error, %s is already set: %s"
+
+static void cliBoardName(char *cmdline)
+{
+    const unsigned int len = strlen(cmdline);
+    if (len > 0 && boardInformationIsSet() && strncmp(getBoardName(), cmdline, len)) {
+        cliPrintLinef(ERROR_MESSAGE, "board_name", getBoardName());
+    } else {
+        if (len > 0) {
+            setBoardName(cmdline);
+            boardInformationUpdated = true;
+        }
+        cliPrintLinef("board_name %s", getBoardName());
+    }
+}
+
+static void cliManufacturerId(char *cmdline)
+{
+    const unsigned int len = strlen(cmdline);
+    if (len > 0 && boardInformationIsSet() && strncmp(getManufacturerId(), cmdline, len)) {
+        cliPrintLinef(ERROR_MESSAGE, "manufactuer_id", getManufacturerId());
+    } else {
+        if (len > 0) {
+            setManufacturerId(cmdline);
+            boardInformationUpdated = true;
+        }
+        cliPrintLinef("manufacturer_id %s", getManufacturerId());
+    }
+}
+
+#if defined(USE_SIGNATURE)
+static void writeSignature(char *signatureStr, uint8_t *signature)
+{
+    for (unsigned i = 0; i < SIGNATURE_LENGTH; i++) {
+        tfp_sprintf(&signatureStr[2 * i], "%02x", signature[i]);
+    }
+}
+
+static void cliSignature(char *cmdline)
+{
+    const unsigned int len = strlen(cmdline);
+
+    char signatureStr[SIGNATURE_LENGTH * 2 + 1] = {0};
+    uint8_t signature[SIGNATURE_LENGTH];
+    if (len > 0) {
+#define BLOCK_SIZE 2
+        for (unsigned i = 0; i < SIGNATURE_LENGTH; i++) {
+            char temp[BLOCK_SIZE + 1];
+            strncpy(temp, &cmdline[i * BLOCK_SIZE], BLOCK_SIZE);
+            temp[BLOCK_SIZE] = '\0';
+            signature[i] = strtoul(temp, NULL, 16);
+        }
+#undef BLOCK_SIZE
+    }
+
```

Comment:
Perhaps there should be some validation that the entered data is valid hexadecimal digits?  For example:
```
# signature this is bad data
signature 00000000ba0dda00000000000000000000000000000000000000000000000000
```
Returns an unexpected and confusing result.